### PR TITLE
glossarywidget: option to whether display newlines in an list

### PR DESCRIPTION
**Pull request by Calvin-Xu**
Part of https://github.com/ripose-jp/Memento/issues/62

### CODE REVIEW

File: src/gui/widgets/definition/glossarywidget.cpp [hunk 1]
```diff
@@ -70,14 +71,26 @@ GlossaryWidget::GlossaryWidget(const size_t          number,
...
+            glos += "</li>";
+        }
+        glos += "</ul>";
+    }
+    else
+    {
+        glos += "<p>";
+        for (QString def : m_def.glossary)
+        {
+            glos += def.replace(QRegExp("\n"), QString("<br>"));
```
Comment (ripose-jp) [inline src/gui/widgets/definition/glossarywidget.cpp:90]: `glos += def.replace('\n', "<br>");`

File: src/gui/widgets/definition/termwidget.cpp [hunk 2]
```diff
@@ -185,7 +186,7 @@ void TermWidget::initUi(const Term &term)
     std::shared_ptr<const AnkiConfig> config = m_client->getConfig();
     for (size_t i = 0; i < term.definitions.size(); ++i)
     {
-        GlossaryWidget *g = new GlossaryWidget(i + 1, term.definitions[i]);
+        GlossaryWidget *g = new GlossaryWidget(i + 1, term.definitions[i], list);
```
Comment (ripose-jp) [inline src/gui/widgets/definition/termwidget.cpp:189]: 80 characters

File: src/gui/widgets/definition/termwidget.h [hunk 3]
```diff
@@ -44,10 +44,12 @@ class TermWidget : public QWidget
      * Constructor for TermWidget.
      * @param term    The term to display. Does not take ownership.
      * @param sources A list of audio sources. Does not take ownership.
+     * @param list    Whether GlossaryWidget displays newlines in an ordered list
```
Comment (ripose-jp) [inline src/gui/widgets/definition/termwidget.h:47]: Add period and 80 characters

File: src/gui/widgets/definition/termwidget.h [hunk 4]
```diff
@@ -101,8 +103,9 @@ private Q_SLOTS:
     /**
      * Puts term information into the UI.
      * @param term The term to populate the UI with.
+     * @param list    Whether GlossaryWidget displays newlines in an ordered list
```
Comment (ripose-jp) [inline src/gui/widgets/definition/termwidget.h:106]: Add period and 80 characters

File: src/gui/widgets/settings/interfacesettings.cpp [hunk 5]
```diff
@@ -158,6 +158,9 @@ void InterfaceSettings::restoreDefaults()
     );
 #endif
 
+    /* Term Display */
+    m_ui->checkGlossaryList->setChecked(SETTINGS_INTERFACE_GLOSSARY_LIST_DEFAULT);
```
Comment (ripose-jp) [inline src/gui/widgets/settings/interfacesettings.cpp:162]: 80 characters

**Comment by Calvin-Xu** [comment]
Thanks. I think these changes should have fixed everything. I've just turned the 80 character margin on (I don't usually use Allman braces so turned off most styling related checks; eyeballing it is bad).

I still feel the option belongs more to interface settings, but I've put it in search settings now. Changing it if you change your mind in the future should be easy.

File: src/gui/widgets/definition/definitionwidget.cpp [hunk 6]
```diff
@@ -99,6 +99,12 @@ void DefinitionWidget::initTheme()
         setStyleSheet(SETTINGS_INTERFACE_DEFINITION_STYLE_DEFAULT);
     }
     settings.endGroup();
+    settings.beginGroup(SETTINGS_SEARCH);
+    m_listGlossary = settings.value(
+            SETTINGS_SEARCH_LIST_GLOSSARY,
```
Comment (ripose-jp) [inline src/gui/widgets/definition/definitionwidget.cpp:104]: This should be in `initSearch`

File: src/gui/widgets/settings/searchsettings.ui [hunk 7]
```diff
@@ -192,12 +192,29 @@ Changing this value from the default is not recommended.</string>
...
        </item>
+       <item>
+        <layout class="QVBoxLayout" name="layoutNewLine">
+         <item>
+          <widget class="QCheckBox" name="checkGlossaryList">
+           <property name="enabled">
+            <bool>true</bool>
+           </property>
+           <property name="text">
+            <string>Show lines of glossary in an unordered list (restart to take effect)</string>
```
Comment (ripose-jp) [inline src/gui/widgets/settings/searchsettings.ui:203]: Don't assume anyone will know what an unordered list is. There is also no reason for this to require a restart to take effect. As long as the correct code is in `initSearch` it shouldn't.

Replace this with
>Use bullet points in term glossaries

**Review by ripose-jp — CHANGES_REQUESTED** [review]

**Comment by Calvin-Xu** [comment]
Changed

**Pull request merged**